src/xcode/ENA/ENA/Source/Models/Task Scheduling/ENATaskScheduler.swift

```diff
@@ -59,20 +59,26 @@ final class ENATaskScheduler {
 		let identifierString = taskIdentifier.backgroundTaskSchedulerIdentifier
 		BGTaskScheduler.shared.register(forTaskWithIdentifier: identifierString, using: .main) { task in
 			self.scheduleTask()
+			let backgroundTask = DispatchWorkItem {
+				execute(task)
+			}
+
 			task.expirationHandler = {
-				logError(message: "ERROR: Task has expired.")
 				self.scheduleTask()
+				backgroundTask.cancel()
```

Comment:
One reminder:
Cancellation does not affect the execution of a work item that has already begun.

Reply:
absolutely right! see my other comment about the DispatchWorkItem for more clarifications